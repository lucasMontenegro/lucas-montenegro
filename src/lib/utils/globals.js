import g from "global"
// renaming "global" to "globals"
// declaring a variable named "global" may cause trouble
export default g