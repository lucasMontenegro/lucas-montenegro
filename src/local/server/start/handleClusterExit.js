const { console } = global
module.exports = function handleClusterExit (worker, code, signal) {
  const { pid } = worker.process
  console.log(`Node cluster worker ${pid} exited: code ${code}, signal ${signal}`)
}