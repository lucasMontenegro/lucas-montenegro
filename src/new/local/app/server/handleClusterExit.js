const { default: globals } = require("new/local/utils/globals")
exports.default = function handleClusterExit (worker, code, signal) {
  const { pid } = worker.process
  globals.console.log(`Node cluster worker ${pid} exited: code ${code}, signal ${signal}`)
}