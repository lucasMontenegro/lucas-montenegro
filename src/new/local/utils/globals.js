// renaming "global" to "globals"
// declaring a variable named "global" may cause troubles
exports.default = require("global")