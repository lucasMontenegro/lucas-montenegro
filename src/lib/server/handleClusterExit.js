import globals from "lib/utils/globals"
export default function handleClusterExit (worker, code, signal) {
  const { pid } = worker.process
  globals.console.log(`Node cluster worker ${pid} exited: code ${code}, signal ${signal}`)
}