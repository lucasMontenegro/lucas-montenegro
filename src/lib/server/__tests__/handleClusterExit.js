import globals from "lib/utils/globals"
import handleClusterExit from "lib/server/handleClusterExit"
const mockFn = jest.fn()
jest.mock(`lib/utils/globals`, () => ({
  __esModule: true,
  default: { console: {} },
}))
globals.console.log = (...args) => mockFn(`console.log`, args)
describe(`lib/server/handleClusterExit`, () => {
  it(`should log`, () => {
    handleClusterExit({ process: { pid: `'worker.process.pid'` } }, `'code'`, `'signal'`)
    expect(mockFn.mock.calls).toMatchSnapshot()
  })
})