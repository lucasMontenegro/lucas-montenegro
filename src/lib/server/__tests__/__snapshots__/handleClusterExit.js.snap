// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/server/handleClusterExit should log 1`] = `
Array [
  Array [
    "console.log",
    Array [
      "Node cluster worker 'worker.process.pid' exited: code 'code', signal 'signal'",
    ],
  ],
]
`;
